azkaban-web-server/src/main/java/azkaban/webapp/AzkabanWebServer.java

```diff
@@ -589,7 +596,9 @@ private void addThreadPoolGauges(final QueuedThreadPool threadPool) {
 
 
   private void startWebMetrics() throws Exception {
-    this.metricsManager.addGauge("WEB-NumQueuedFlows", this.executorManager::getQueuedFlowSize);
+    this.metricsManager
+        .addGauge("WEB-NumQueuedFlows", ((ExecutorManager) this
```

Comment:
I hope this is temporary, predicated on the above `todo`. Obviously, this cast will fail when `executorManagerAdapter` is an `ExecutionController`. Is this just to postpone `addGauge` refactoring to accept ExecutorManagerAdapter instead of ExecutorManager?

Reply:
Yes, you are right. I have left the metrics/jmx part for later implementation since there are quite some differences between the new module and the old one, which would require a separate PR change.